src/plugins/irc-events/ison.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const Msg = require("../../models/msg");
+const Chan = require("../../models/chan");
+
+module.exports = function(irc) {
+	const client = this;
+	irc.on("users online", (data) => {
+		if (data.nicks.length === 1) {
+			const target = client.find(client.lastActiveChannel);
```

Comment:
This is gonna be weird with multiple clients open. As we discussed on IRC, the better way is most likely querying all names with `ison` at once.

Reply:
and just compare the array of open queries with the returned array and find what's missing?

Reply:
Pretty much, yeah.